aesara/link/numba/dispatch.py

```diff
@@ -1865,3 +1877,94 @@ def batched_dot(x, y):
 # NOTE: The remaining `aesara.tensor.blas` `Op`s appear unnecessary, because
 # they're only used to optimize basic `Dot` nodes, and those GEMV and GEMM
 # optimizations are apparently already performed by Numba
+
+
+def make_numba_random_fn(node, np_random_fn, np_input_names, np_random_fn_name):
+    tuple_size = get_vector_length(node.inputs[1])
+    size_dims = tuple_size - max(i.ndim for i in node.inputs[3:])
+
+    input_names = ", ".join(
+        ["rng", "size", "dtype"] + [get_name_for_object(i) for i in node.inputs[3:]]
+    )
+
+    # If size is None or () we return a singular value directly
+    if tuple_size == 0:
+        global_env = {"np": np, np_random_fn_name: np_random_fn}
+
+        random_fn_src = f"""
+def random_variable({input_names}):
+    return (rng, {np_random_fn_name}({np_input_names}))
+        """
+    else:
+        # Or, create a Numba JITable function that implements the `size` parameter
+        global_env = {
+            "np": np,
+            np_random_fn_name: np_random_fn,
+            "to_fixed_tuple": to_fixed_tuple,
+            "tuple_size": tuple_size,
+            "size_dims": size_dims,
+        }
+        random_fn_src = f"""
+def random_variable({input_names}):
+    size = to_fixed_tuple(size, tuple_size)
+    data = np.empty(size)
+    for i in np.ndindex(size[:size_dims]):
+        data[i] = {np_random_fn_name}({np_input_names})
+    return (rng, data)
+    """
+    random_fn = compile_function_src(random_fn_src, "random_variable", global_env)
+    random_fn = numba.njit(random_fn)
+
+    return random_fn
+
+
+@numba_funcify.register(RandomVariable)
+def numba_funcify_RandomVariable(op, node, **kwargs):
+
+    # XXX TODO: The following approach will only work for strict NumPy
+    # pass-through `Op`s.
+
+    name = op.name
+    np_random_func = getattr(np.random, name)
+
+    # Make a broadcast-capable version of the Numba supported scalar sampling
+    # function
+    np_names = [get_name_for_object(i) for i in node.inputs[3:]]
+    np_input_names = ", ".join(np_names)
+    np_global_env = {
+        "np_random_func": np_random_func,
+        "numba_vectorize": numba.vectorize,
+    }
+    np_random_fn_name = f"aesara_random_{get_name_for_object(np_random_func)}"
+    np_random_fn_src = f"""
+@numba_vectorize
+def {np_random_fn_name}({np_input_names}):
+    return np_random_func({np_input_names})
+    """
+    np_random_fn = compile_function_src(
+        np_random_fn_src, np_random_fn_name, np_global_env
+    )
+
+    return make_numba_random_fn(node, np_random_fn, np_input_names, np_random_fn_name)
+
+
+@numba_funcify.register(HalfNormalRV)
```

Comment:
For now, we can get away with this way of implementing non numpy RV's but it'll be nice to have a 'object mode' to fall back to if this fails. (Since, this particular implementation assumes we can convert any RV from `scipy.stats` to it's corresponding Numpy/Numba version, but that might not be always the case)

Reply:
Exactly!